lxd/provider.go

```diff
@@ -160,7 +160,16 @@ func Provider() *schema.Provider {
 				Optional:    true,
 				Description: descriptions["lxd_config_dir"],
 				DefaultFunc: func() (interface{}, error) {
-					return os.ExpandEnv("$HOME/.config/lxc"), nil
+					// Check for the presence of the /var/snap/lxd directory. If the
+					// directory exists, return snap's config path, otherwise return
+					// the fallback path.
+
+					_, err := os.Stat("/var/snap/lxd")
+					if os.IsNotExist(err) {
+						return os.ExpandEnv("$HOME/.config/lxc"), nil
+					}
+
+					return os.ExpandEnv("$HOME/snap/lxd/common/config"), nil
```

Comment:
The comment doesn't match the code. Checking for the dir first should lead to a smaller diff like:

```
    // Check for the presence of the /var/snap/lxd directory. If the
    // directory exists, return snap's config path, otherwise return
    // the fallback path.

    _, err := os.Stat("/var/snap/lxd")
    if os.IsExist(err) {
	return os.ExpandEnv("$HOME/snap/lxd/common/config"), nil
    }

    return os.ExpandEnv("$HOME/.config/lxc"), nil
```

Reply:
I agree, will fix that one.

Reply:
Just to clarify the change, the `os.IsExist` is not sufficient, since this checks whether the error is of type *file already exists* or *directory not empty*. If there is no error, `os.IsExist` will return false, so we also have to check whether the `err == nil`.

Reply:
Ah, right, thanks!